bundles/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,485 @@
+        Future<?> localRef = forecastUpdater;
+        if (localRef != null) {
+            localRef.cancel(false);
+        }
+        localRef = instantUpdate;
+        if (localRef != null) {
+            localRef.cancel(false);
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        if (!isModifyingCurrentConfig(configurationParameters)) {
+            return;
+        }
+
+        validateConfigurationParameters(configurationParameters);
+
+        Configuration configuration = editConfiguration();
+        for (Map.Entry<String, Object> configurationParameter : configurationParameters.entrySet()) {
+            configuration.put(configurationParameter.getKey(), configurationParameter.getValue());
+        }
+
+        if (isInitialized()) {
+            updateConfiguration(configuration);
+            config = configuration.as(SmhiConfiguration.class);
+            updateThing(editThing().withChannels(createChannels()).build());
+            if (!isPollingStarted()) {
+                startPolling();
+            }
+            updateNow();
+        } else {
+            // persist new configuration and notify Thing Manager
+            updateConfiguration(configuration);
+            ThingHandlerCallback callback = getCallback();
+            if (callback != null) {
+                callback.configurationUpdated(this.getThing());
+            } else {
+                logger.warn("Handler {} tried updating its configuration although the handler was already disposed.",
+                        this.getClass().getSimpleName());
+            }
+        }
+    }
+
+    /**
+     * Update channels with new forecast data.
+     * 
+     * @param timeSeries A {@link TimeSeries} object containing forecasts.
+     */
+    private void updateChannels(TimeSeries timeSeries) {
+        // Loop through hourly forecasts and update those available
+        for (int i = 0; i < 25; i++) {
+            List<Channel> channels = thing.getChannelsOfGroup("hour_" + i);
+            if (channels.isEmpty()) {
+                continue;
+            }
+            Forecast forecast = timeSeries.getForecast(i);
+            if (forecast != null) {
+                channels.forEach(c -> {
+                    String id = c.getUID().getIdWithoutGroup();
+                    BigDecimal value = forecast.getParameter(id);
+                    updateChannel(c, value);
+                });
+            }
+        }
+        // Loop through daily forecasts and updates those available
+        for (int i = 0; i < 10; i++) {
+            List<Channel> channels = thing.getChannelsOfGroup("day_" + i);
+            if (channels.isEmpty()) {
+                continue;
+            }
+            Forecast forecast = timeSeries.getForecast(currentDay, 24 * i + 12);
+
+            if (forecast == null) {
+                logger.debug("No forecast yet for {}", currentDay.plusHours(24 * i + 12));
```

Comment:
You are doing a small calculation in you debug statement, I don't know how often this path is executed, but people without debug logging still do the calculation each time. You could guard it with `logger.isDebugEnabled()`

Reply:
Is it just the integer calculation you're concerned about or the plusHours() as well? I could do the calculation separately and use it both for `timeSeries.getForecast()` and the debug statement (would look better DRY-wise). Anyway, this logging statement shouldn't appear more than once an hour at most.

Reply:
Especially the `plusHours` because that is kinda black box, the integer calculation should be done in a single CPU tick :-)

Given its frequency it's not problematic